### PR TITLE
[UI] Add telemetry base card

### PR DESCRIPTION
**Pull request by JessicaDJ0807**
Telemetry base card is implemented in this PR.
- Change data refresh interval format
- Create TelemetryCardContent Component
- Create TelemetryCard Component
- Implement TelemetryDashboard Component

The original telemetry page looks like: https://screenshot.googleplex.com/Eu8rVJEqLqaHXzu.png, and with base cards, it looks like: https://screenshot.googleplex.com/Bwrrpth62HnrPY4.png.

BUG: b/295126796

### CODE REVIEW

File: diagnostics-app/src/app/telemetry/telemetry-dashboard/telemetry-dashboard.component.css
Comment (BaiiYuan) [inline diagnostics-app/src/app/telemetry/telemetry-dashboard/telemetry-dashboard.component.css:2]: Don't change the copyright year. See [go/copyright](https://goto.google.com/copyright).

Comment (JessicaDJ0807) [reply diagnostics-app/src/app/telemetry/telemetry-dashboard/telemetry-dashboard.component.css:2]: Thanks for the reminder. I changed it following the rule.

File: diagnostics-app/src/app/telemetry/telemetry-dashboard/telemetry-dashboard.component.ts
Comment (BaiiYuan) [inline diagnostics-app/src/app/telemetry/telemetry-dashboard/telemetry-dashboard.component.ts:32]: Added comments to describe these.

One nit: This might be out of this PR. `DpslTypes` should be renamed to `TelemetryUnion`, which is clearer.

File: diagnostics-app/src/app/telemetry/telemetry-dashboard/telemetry-dashboard.component.ts
Comment (BaiiYuan) [inline diagnostics-app/src/app/telemetry/telemetry-dashboard/telemetry-dashboard.component.ts:58]: Check if  `displayType` is in `refreshIntervals`.

File: diagnostics-app/src/app/telemetry/telemetry-dashboard/telemetry-dashboard.component.ts
Comment (BaiiYuan) [inline diagnostics-app/src/app/telemetry/telemetry-dashboard/telemetry-dashboard.component.ts:23]: `Display` is also one category of Telemetry, which could be confused. Maybe simply use `_telemetryTypes` and add a comment for it.

File: diagnostics-app/src/app/telemetry/telemetry-dashboard/telemetry-dashboard.component.html
Comment (BaiiYuan) [inline diagnostics-app/src/app/telemetry/telemetry-dashboard/telemetry-dashboard.component.html:2]: Hmmm I think it's better to retain earliest copyright year only.

File: diagnostics-app/src/app/telemetry/telemetry-dashboard/telemetry-dashboard.component.ts
Comment (BaiiYuan) [reply diagnostics-app/src/app/telemetry/telemetry-dashboard/telemetry-dashboard.component.ts:32]: What is these variable? What are they used for?

File: diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card-content/telemetry-card-content.component.html
Comment (BaiiYuan) [inline diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card-content/telemetry-card-content.component.html:3]: Random idea: When will `item.value` != `object`? If it is loading, maybe we can create a loading template for the loading items instead of skipping them.

File: diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card-content/telemetry-card-content.component.ts
Comment (BaiiYuan) [inline diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card-content/telemetry-card-content.component.ts:20]: It seems that it is used for nested content and not used now?
Add a comment for it or removed it.

File: diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card.component.ts
Comment (BaiiYuan) [inline diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card.component.ts:19]: Could we use `undefined` for enum?

**Comment by ycheng627** [comment]
High level comment,

I think we can separate this into two different PRs.

First PR can be for all changes in telemetry service, deleting unneeded functions, etc. For compatibility reasons, I think we can delete the CPU, memory, storage card here. Its okay for the telemetry dashboard to be blank after this PR is merged.

Second PR can be for all changes in the actual base card.

File: diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card.component.ts
Comment (BaiiYuan) [inline diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card.component.ts:27]: Could we have an optional error type and use it to check if any error is happened?

File: diagnostics-app/src/app/core/config/data-refresh-intervals.ts
Comment (BaiiYuan) [inline diagnostics-app/src/app/core/config/data-refresh-intervals.ts:8]: Use standalone variable for default value. e.g. `defaultTelemetryRefreshInterval`

File: diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card.component.ts
Comment (BaiiYuan) [inline diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card.component.ts:25]: It is not a good idea to use non-null assertion here since `_info` could be null if the error is happened.

File: diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card.component.ts
Comment (BaiiYuan) [inline diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card.component.ts:56]: Setup error for unexpected behavior.
```
} else {
  // Setup error.
}
```

File: diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card-content/telemetry-card-content.component.css
Comment (ycheng627) [inline diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card-content/telemetry-card-content.component.css:7]: Add comment about card-content being recursively called, so we add some left padding for indentation.

**Review by ycheng627 — COMMENTED** [review]
Overall LGTM, just some small nits and questions here and there.

File: diagnostics-app/src/app/telemetry/telemetry-dashboard/telemetry-dashboard.component.ts
Comment (JessicaDJ0807) [reply diagnostics-app/src/app/telemetry/telemetry-dashboard/telemetry-dashboard.component.ts]: Done.

File: diagnostics-app/src/app/telemetry/telemetry-dashboard/telemetry-dashboard.component.ts
Comment (JessicaDJ0807) [reply diagnostics-app/src/app/telemetry/telemetry-dashboard/telemetry-dashboard.component.ts]: It is renamed to _visibleTypes.

File: diagnostics-app/src/app/telemetry/telemetry-dashboard/telemetry-dashboard.component.html
Comment (JessicaDJ0807) [reply diagnostics-app/src/app/telemetry/telemetry-dashboard/telemetry-dashboard.component.html]: Done.

File: diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card-content/telemetry-card-content.component.html
Comment (JessicaDJ0807) [reply diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card-content/telemetry-card-content.component.html]: If the item.value is not an object, it displays the value directly.

File: diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card-content/telemetry-card-content.component.ts
Comment (JessicaDJ0807) [reply diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card-content/telemetry-card-content.component.ts]: It is used for rendering nested object, so it is used now.

File: diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card.component.ts
Comment (JessicaDJ0807) [reply diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card.component.ts]: The value of 'type' will be undefined for the very first moment, so the template will check if 'type' is equal to undefined, it will not render the content. The input is also set to be required.

File: diagnostics-app/src/app/core/services/telemetry.service.ts
Comment (JessicaDJ0807) [reply diagnostics-app/src/app/core/services/telemetry.service.ts]: The old methods are removed.

File: diagnostics-app/src/app/core/services/telemetry.service.ts
Comment (JessicaDJ0807) [reply diagnostics-app/src/app/core/services/telemetry.service.ts]: The <code>fetchInfo</code> function is removed. And the <code>fetchTelemetryData</code> function is changed to be public so that UI can call it to get the requested Telemetry data.

File: diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card-content/telemetry-card-content.component.html
Comment (JessicaDJ0807) [reply diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card-content/telemetry-card-content.component.html]: Done.

File: diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card-content/telemetry-card-content.component.ts
Comment (JessicaDJ0807) [reply diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card-content/telemetry-card-content.component.ts]: The input <code>info</code> will be undefined for the first moment even if the value is inputted correctly. And if it is undefined, the frontend will show nothing.

File: diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card.component.html
Comment (JessicaDJ0807) [reply diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card.component.html]: Done.

File: diagnostics-app/src/app/telemetry/telemetry-dashboard/telemetry-dashboard.component.ts
Comment (JessicaDJ0807) [reply diagnostics-app/src/app/telemetry/telemetry-dashboard/telemetry-dashboard.component.ts]: Comments for variables are added. And a bug is filed for renaming DpslTypes.

File: diagnostics-app/src/app/core/config/data-refresh-intervals.ts
Comment (JessicaDJ0807) [reply diagnostics-app/src/app/core/config/data-refresh-intervals.ts]: Done.

File: diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card.component.ts
Comment (JessicaDJ0807) [reply diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card.component.ts:25]: If <code>_info</code> is undefined, there are two possible circumstances:
1. The info is loading and no error occurs, then a loading spinner will be shown
2. An error occurs, then it will show an error message as the card content.

File: diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card.component.ts
Comment (JessicaDJ0807) [reply diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card.component.ts:58]: Done. An error will be thrown.

File: diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card-content/telemetry-card-content.component.ts
Comment (JessicaDJ0807) [reply diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card-content/telemetry-card-content.component.ts]: I added a comment to explain the use of the function. And I think it's better to maintain the original order since it conforms to what users will expect to understand the data easily.

File: diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card-content/telemetry-card-content.component.ts
Comment (JessicaDJ0807) [reply diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card-content/telemetry-card-content.component.ts]: Done.

File: diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card.component.ts
Comment (JessicaDJ0807) [reply diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card.component.ts:33]: A bug is filed for renaming the private properties.

File: diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card-content/telemetry-card-content.component.html
Comment (JessicaDJ0807) [reply diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card-content/telemetry-card-content.component.html:7]: I don't think we can typecast it into an object. The <code>$any()</code> is just for accessing the methods of <code>item</code>.

Comment (andy920262) [reply diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card-content/telemetry-card-content.component.html:7]: Will it cause any problem if we declare `info` default to an empty object instead of `undefined`?

File: diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card.component.ts
Comment (andy920262) [inline diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card.component.ts:26]: `private _hasError = false;` already infers the type. Don't need type annotation here.

File: diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card.component.spec.ts
Comment (andy920262) [inline diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card.component.spec.ts:20]: Consider adding some test case for error occurred

File: diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card-content/telemetry-card-content.component.html
Comment (JessicaDJ0807) [reply diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card-content/telemetry-card-content.component.html:7]: It returns an error when building:
```
The last overload gave the following error.
Argument of type 'Object' is not assignable to parameter of type 'Record<string, string> | ReadonlyMap<string, string> | null | undefined'.
The 'Object' type is assignable to very few other types. Did you mean to use the 'any' type instead? 
```

File: diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card.component.ts
Comment (JessicaDJ0807) [reply diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card.component.ts]: After discussing with my hosts, the variable <code>_hasError</code> will be removed. The other variable <code>_error</code> will be used for checking if any error has occurred and caching the error message.

File: diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card.component.ts
Comment (JessicaDJ0807) [reply diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card.component.ts]: Done!

File: diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card.component.spec.ts
Comment (ycheng627) [reply diagnostics-app/src/app/telemetry/telemetry-card/telemetry-card.component.spec.ts:20]: For the scope of the intern project and with limited timeframe, I think think we will keep it to just feature development.

Unit test will be refactored and added by me in another time.

**Review by ycheng627 — APPROVED** [review]
LGTM. Wait for other's final approval

**Review by BaiiYuan — APPROVED** [review]

**Comment by BaiiYuan** [comment]
LGTM. Thanks for the update.

**Pull request merged**
